src/java/org/apache/cassandra/index/sai/memory/KeyRangeIterator.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.memory;
+
+import java.io.IOException;
+import java.util.PriorityQueue;
+import java.util.SortedSet;
+
+import org.apache.cassandra.index.sai.utils.PrimaryKey;
+import org.apache.cassandra.index.sai.utils.RangeIterator;
+
+public class KeyRangeIterator extends RangeIterator
```

Comment:
nit: Worth documenting this as `@ThreadSafe`?

Reply:
I don't think this class is threadsafe or needs to be. It is used for single queries that are always going to be on one thread. As in, this is never going to be shared between threads. Or, did you mean `@NotThreadSafe`?